frontend/src/api/prometheus/__tests__/useQueryRangeResourceData.spec.ts

```diff
@@ -0,0 +1,102 @@
+import axios from 'axios';
+import { act } from '@testing-library/react';
+import { TimeframeTitle } from '~/pages/modelServing/screens/types';
+import { testHook } from '~/__tests__/unit/testUtils/hooks';
+import { mockPrometheusServing } from '~/__mocks__/mockPrometheusServing';
+import useQueryRangeResourceData from '~/api/prometheus/useQueryRangeResourceData';
+
+jest.mock('axios', () => ({
+  post: jest.fn(),
+}));
+
+const mockAxios = jest.mocked(axios.post);
+
+describe('useQueryRangeResourceData', () => {
+  const responsePredicate = jest.fn();
+  const query = 'testQuery';
+  const end = 123456;
+  const namespace = 'testNamespace';
+
+  const timeframe: TimeframeTitle = TimeframeTitle.ONE_HOUR;
+
+  it('should fetch data successfully and handle refresh ', async () => {
+    mockAxios.mockResolvedValue({ data: { response: mockPrometheusServing({}) } });
```

Comment:
instead of calling mock function multiple time , you can store it in a const and reuse it